sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -172,6 +172,17 @@ class Dataset[T] private[sql](
     this(sqlContext.sparkSession, logicalPlan, encoder)
   }
 
+  /** A friendly name for this Dataset */
+  @Since("2.2.0")
```

Comment:
this doesn't seem like the convention for doc string. For example
```
   /**
    * Returns a Java list that contains randomly split Dataset with the provided weights.
    *
    * @param weights weights for splits, will be normalized if they don't sum to 1.
    * @param seed Seed for sampling.
    *
    * @group typedrel
    * @since 2.0.0
    */
```

it should have param, group, since.